spec/lib/sarif/language_version/language_version_scanner_sarif_spec.rb

```diff
@@ -0,0 +1,100 @@
+RSpec.shared_examples_for "language version scanner sarif" do |scanner_class,
+                                                               scanner_name,
+                                                               valid_version_repo_path,
+                                                               invalid_version_repo_path,
+                                                               error_msg|
+  describe '#build_runs_object' do
+    let(:config_file) { YAML.load_file(invalid_version_repo_path + '/salus.yml') }
+    let(:scanner) do
+      scanner_class.new(repository: repo,
+                    config: config_file['scanner_configs'][scanner_name.to_s])
+    end
+    before { scanner.run }
+
+    context 'version mismatch found in project' do
+      let(:path) { invalid_version_repo_path }
+      let(:repo) { Salus::Repo.new(path) }
+
+      it 'creates valid sarif report with results populated' do
+        adapter = described_class.new(scanner.report, path)
+        report = adapter.build_runs_object(true)
+        rules = report['tool'][:driver]['rules']
+        results = report['results']
+
+        expect(results.size).to eq(1)
+        expect(results).to include(
+          {
+            "ruleId": "LV0001",
+            "ruleIndex": 0,
+            "level": "error",
+            "message": {
+              "text": error_msg
+            },
+            "properties": { "severity": "HIGH" },
+            "locations": [
+              {
+                "physicalLocation": {
+                  "artifactLocation": {
+                    "uri": "",
+                    "uriBaseId": "%SRCROOT%"
+                  }
+                }
+              }
+            ]
+          }
+        )
+
+        doc = "https://github.com/coinbase/salus/blob/master/docs/scanners/language_version_scan.md"
```

Comment:
I'm not worried about being DRY in a spec... you could reference the CONST here if you wanted to.

Reply:
Updated.